File: leetcode_js/3Sum Closest.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var threeSumClosest = function(nums, target) {
    nums.sort(function(a,b){return a - b;})
    var res = nums[0] + nums[1] + nums[2];
    for(var i = 0; i < nums.length -2; i++){
    	var le = i + 1, ri = nums.length - 1;
    	while(le < ri){
    		var tmp = nums[i] + nums[le] + nums[ri];
    		if(tmp < target){
    			le++;
    		}else if(tmp > target){
    			ri--;
    		}else{
    			return target;
    		}
    		if(Math.abs(target - tmp) <
    		 Math.abs(res - target)){
    			res = tmp;
    		}
    	}
    }
    return res;
};

File: leetcode_java/3sum-closest/3SumClosest.java
public class Solution {
    public int threeSumClosest(int[] nums, int target) {
        int min = Integer.MAX_VALUE;
        int result = 0;
        int len = nums.length;
        int j ,k ;
        
        int diff;
        int sum = 0;
        Arrays.sort(nums);
        for(int i  = 0; i < len ;i++){
            j = i + 1;
            k = len -1 ;
            while(j < k){
                sum = nums[i] + nums[j] + nums[k];
                diff = Math.abs(sum - target);
                if(diff == 0){
                    return sum;
                }
                if(diff < min){
                    min = diff;
                    result = sum;
                }
                if(sum < target){
                    j++;
                }else{
                    k--;
                }
            }
        }
        return result;
    }
}